test/unit/gateways/payeezy_test.rb

```diff
@@ -279,8 +279,8 @@ def test_successful_general_credit_with_soft_descriptors
     response = stub_comms do
       @gateway.credit(@amount, @credit_card, @options.merge(@options_mdd))
     end.check_request do |_endpoint, data, _headers|
-      json = '{"transaction_type":"refund","currency_code":"USD","amount":"100","method":"credit_card","credit_card":{"type":"Visa","cardholder_name":"Longbob Longsen","card_number":"4242424242424242","exp_date":"0922","cvv":"123"},"soft_descriptors":{"dba_name":"Caddyshack","street":"1234 Any Street","city":"Durham","region":"North Carolina","mid":"mid_1234","mcc":"mcc_5678","postal_code":"27701","country_code":"US","merchant_contact_info":"8885551212"},"merchant_ref":null}'
-      assert_match json, data
+      soft_descriptors_regex = %r("soft_descriptors":{"dba_name":"Caddyshack","street":"1234 Any Street","city":"Durham","region":"North Carolina","mid":"mid_1234","mcc":"mcc_5678","postal_code":"27701","country_code":"US","merchant_contact_info":"8885551212"})
+      assert_match soft_descriptors_regex, data
```

Comment:
These two tests were failing because they contained the credit card year in the json we were matching against.

I simplified the assertions to only match against the portions of the request that we claim to be testing.

Reply:
Thanks, this looks much cleaner.